src/elm/4_input/4_input.md

```diff
@@ -0,0 +1,288 @@
+---
+title: "Input fra brukeren"
+level: 3
+language: nb-NO
+author: Alexander Perry
+---
+
+# Introduksjon {.intro}
+
+Input-felter lar brukere skrive inn tekst, tall, datoer og så videre i nettsiden vår. Informasjonen vi får fra brukerne kan vi endre, skrive ut, og kombinere med annen informasjon.
+
+# Hente tekst fra brukeren {.activity}
+
+## Moduler og program {.check}
+
+- Øverst i koden vår må vi si hvilke *moduler* vi skal bruke
+
+```elm
+import Html exposing (Html, Attribute, beginnerProgram, text, div, input, button)
+import Html.Attributes exposing (..)
+import Html.Events exposing (onInput, onClick)
+import String
+```
+
+## Modell, meldinger og oppdatering {.check}
+
+- Vi oppretter programmet vårt med en *modell*, et *view* og en *update*-funksjon. Vi kommer tilbake til hvordan vi oppretter disse, skriv inn følgende i koden:
+
+```elm
+main =
+  beginnerProgram { model = "", view = view, update = update }
+```
+
+- Så trenger vi å si hva slags *meldinger* (`Msg`) vi skal sende i programmet. Til å begynne med har vi bare én meldingstype, og det er meldingen som sendes når brukeren endrer teksten i input-feltet, her kaller vi den for `Endre`.
+
+```elm
+type Msg = Endre String
+```
+
+- Etter det må vi skrive hva som skal skje når teksten oppdateres med en *update*-funksjon
+
+```elm
+update msg model =
+  case msg of
+    Endre tekst ->
+      tekst
+```
+
+Nå hentes bare teksten inn, og sendes rett ut igjen. Det er kanskje litt rart, når funksjonen vår heter `Endre`? La oss endre teksten!
+
+For å endre teksten, kan vi bruke innebygde funksjoner i `String`-modulen.
+
+## Skriv ut teksten {.check}
+- Til slutt må vi ha et sted som viser teksten, da bruker vi et *view*
+
+```elm
+view model =
+  div []
+    [ input [ placeholder "Baklengstekst", onInput Endre ] []
+    , div [ ] [ text model ]
+    ]
+```
+
+`input [ placeholder "Baklengstekst", onInput Endre ] []` er det som blir til et input-felt. Teksten etter `placeholder` sier hva som skal stå i feltet før brukeren skriver inn noe. `onInput` sier hva som skal skje når brukeren skriver inn tekst. I vårt tilfelle sender vi en `Endre`-melding som hentes inn i `update`-funksjonen. Når brukeren skriver inn "Teodor", sendes en melding av typen `Endre "Teodor"` inn i `update`-funksjonen.
+
+Setter vi alle delene sammen, vil koden vår se slik ut:
+
+```elm
+import Html exposing (Html, Attribute, beginnerProgram, text, div, input)
+import Html.Attributes exposing (..)
+import Html.Events exposing (onInput)
+import String
+
+
+main =
+  beginnerProgram { model = "", view = view, update = update }
+
+
+-- UPDATE
```

Comment:
Protip er lagt til: "I koden over har vi skrevet `-- UPDATE` og `-- VIEW`. Det har vi gjort for å gjøre koden mer strukturert og lesbar. Alle linjer som starter med to bindestreker (--) ignoreres i Elm. Dette er hva vi kaller kommentarer! Kommentarer er ment for menneskene som leser koden, enten deg selv eller de du samarbeider med."